code/controllers/subsystem/html_audio.dm

```diff
@@ -0,0 +1,274 @@
+SUBSYSTEM_DEF(html_audio)
+	name = "HTML Audio"
+	init_order = INIT_ORDER_HTMLAUDIO
+	flags = SS_NO_FIRE
+	var/max_channels = 512 // set this to how many total channels you want
+	var/list/channel_assignment = list() // list([atom in the world], [atom in the world], ...) for keeping track of what channels are in use and by what
+	var/list/channel_loop_status = list() // list([atom in the world] = TRUE/FALSE, ...) for keeping track of what channels are looping
+	var/list/channel_tts_status = list() // list([atom in the world] = TRUE/FALSE, ...) for keeping track of what channels are TTS
+	var/list/listeners = list() // list of client listeners to update when audio gets added
+	var/list/active_urls = list() // list("url_here", ...), ref'd by active_urls[channel_id]
```

Comment:
All of these should be `///` before the comment or `//!` where they are now so that they can act as doc comments